### PR TITLE
chore(broker): add new VariableDocumentRecord

### PR DESCRIPTION
**Pull request by npepinpe**
- adds new VariableDocumentRecord representing a JSON document of key/value pairs
- adds processor for single UPDATE command with support for different update semantics (local/propagate)

### CODE REVIEW

**Comment by npepinpe** [comment]
Related to #1617 , part 1 of...idk, 3? As far as I'm concerned, that issue is well described, I'm just splitting this into more than one PR to make it easier on the reviewer.

This PR simply adds a new record type to handle updating a particular variable scope with local/propagating semantics based on a document.

Next PRs will add functionality:

1. Export to Elasticsearch
1. Wire `UpdateWorkflowInstancePayloadRequest` to new `VariableDocumentRecord#UPDATE` command and remove `WorkflowInstanceRecord#PAYLOAD_UPDATE`
1. Rename `UpdateWorkflowInstancePayload` command to `SetVariables` and update clients

File: broker-core/src/test/java/io/zeebe/broker/workflow/processor/variable/UpdateVariableDocumentProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        .setScopeKey(instance.getKey())
+        .setUpdateSemantics(VariableDocumentUpdateSemantic.PROPAGATE)
+        .setDocument(document);
+
+    // when
+    processor.onCommand(command, controller);
+
+    // then
+    assertAccepted(command);
+    verify(variablesState, times(1)).setVariablesFromDocument(instance.getKey(), document);
```
Comment (Zelldon) [inline broker-core/src/test/java/io/zeebe/broker/workflow/processor/variable/UpdateVariableDocumentProcessorTest.java]: I think you missed here an assertion, that the updated record is written?

File: test-util/src/main/java/io/zeebe/test/util/collection/Maps.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.test.util.collection;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Maps {
+  public static <K, V> Map<K, V> of(Map.Entry<K, V>... entries) {
```
Comment (Zelldon) [inline test-util/src/main/java/io/zeebe/test/util/collection/Maps.java]: Would be nice if we could some newer java lang features heh? :sweat_smile:  like https://www.techiedelight.com/initialize-map-java9/

**Review by ChrisKujawa — APPROVED** [review]
Well done :+1: can you take a look at my comment above and maybe fix it, after that should be good to merge.

File: broker-core/src/test/java/io/zeebe/broker/workflow/processor/variable/UpdateVariableDocumentProcessorTest.java (hunk 1 above)
Comment (npepinpe) [reply broker-core/src/test/java/io/zeebe/broker/workflow/processor/variable/UpdateVariableDocumentProcessorTest.java]: It's easy to miss but it uses two different methods on the `variablesState`, one is `setVariablesFromDocument` and the other is `setVariablesLocalFromDocument`

File: test-util/src/main/java/io/zeebe/test/util/collection/Maps.java (hunk 2 above)
Comment (npepinpe) [reply test-util/src/main/java/io/zeebe/test/util/collection/Maps.java]: :see_no_evil:

**Pull request merged**